src/integer_mod_q/mat_zq/get.rs

```diff
@@ -26,139 +27,43 @@ use flint_sys::{
 use std::{fmt::Display, mem::MaybeUninit};
 
 impl MatZq {
-    /// Returns the modulus of the matrix as a [`Modulus`].
+    /// Create a [`MatZ`] from a [`MatZq`].
     ///
-    /// # Examples
-    /// ```
-    /// use qfall_math::integer_mod_q::MatZq;
+    /// The output matrix entries values will be in the range `[0, Modulus)`.
+    /// Use [`MatZq::get_closest_to_zero_representative`] if they should be
+    /// close to `0`.
     ///
-    /// let matrix = MatZq::new(5, 10, 7);
-    /// let entry = matrix.get_mod();
-    /// ```
-    pub fn get_mod(&self) -> Modulus {
-        self.modulus.clone()
-    }
-}
-
-impl GetNumRows for MatZq {
-    /// Returns the number of rows of the matrix as a [`i64`].
+    /// Returns the matrix as a [`MatZ`].
     ///
     /// # Examples
     /// ```
+    /// use qfall_math::integer::MatZ;
     /// use qfall_math::integer_mod_q::MatZq;
-    /// use qfall_math::traits::*;
-    ///
-    /// let matrix = MatZq::new(5, 6, 7);
-    /// let rows = matrix.get_num_rows();
-    /// ```
-    fn get_num_rows(&self) -> i64 {
-        self.matrix.mat[0].r
-    }
-}
-
-impl GetNumColumns for MatZq {
-    /// Returns the number of columns of the matrix as a [`i64`].
-    ///
-    /// # Examples
-    /// ```
-    /// use qfall_math::integer_mod_q::MatZq;
-    /// use qfall_math::traits::*;
-    ///
-    /// let matrix = MatZq::new(5, 6, 7);
-    /// let rows = matrix.get_num_columns();
-    /// ```
-    fn get_num_columns(&self) -> i64 {
-        self.matrix.mat[0].c
-    }
-}
-
-impl GetEntry<Z> for MatZq {
-    /// Outputs the [`Z`] value of a specific matrix entry.
-    ///
-    /// Parameters:
-    /// - `row`: specifies the row in which the entry is located
-    /// - `column`: specifies the column in which the entry is located
-    ///
-    /// Negative indices can be used to index from the back, e.g., `-1` for
-    /// the last element.
-    ///
-    /// Returns the [`Z`] value of the matrix at the position of the given
-    /// row and column or an error, if the number of rows or columns is
-    /// greater than the matrix or greater than the matrix.
-    ///
-    /// # Examples
-    /// ```rust
-    /// use qfall_math::integer_mod_q::MatZq;
-    /// use qfall_math::traits::GetEntry;
-    /// use qfall_math::integer::Z;
     /// use std::str::FromStr;
     ///
-    /// let matrix = MatZq::from_str("[[1, 2, 3],[4, 5, 6],[7, 8, 9]] mod 10").unwrap();
+    /// let m = MatZq::from_str("[[1, 2],[3, -1]] mod 5").unwrap();
     ///
-    /// assert_eq!(Z::from(3), matrix.get_entry(0, 2).unwrap());
-    /// assert_eq!(Z::from(8), matrix.get_entry(2, 1).unwrap());
-    /// assert_eq!(Z::from(8), matrix.get_entry(-1, -2).unwrap());
+    /// let a = m.get_mat();
```

Comment:
maybe add an assertion with the expected matrix so the person looking at the example can verify that the -1 will be 'changed' to a 4